crates/sui-indexer-alt-graphql/src/api/types/validator_set.rs

```diff
@@ -1,26 +1,36 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use async_graphql::SimpleObject;
-use sui_types::sui_system_state::sui_system_state_inner_v1::ValidatorSetV1;
+use async_graphql::connection::{Connection, CursorType, Edge};
+use async_graphql::{ComplexObject, Context, SimpleObject};
+use sui_types::sui_system_state::sui_system_state_inner_v1::{ValidatorSetV1, ValidatorV1};
 
 use crate::{
+    api::scalars::cursor::JsonCursor,
     api::{
         scalars::{big_int::BigInt, sui_address::SuiAddress},
         types::validator::Validator,
     },
+    error::RpcError,
+    pagination::{Page, PaginationConfig},
     scope::Scope,
 };
 
+pub(crate) type CValidator = JsonCursor<u64>;
+
 /// Representation of `0x3::validator_set::ValidatorSet`.
-#[derive(Clone, Debug, SimpleObject, Default)]
+#[derive(Clone, Debug, SimpleObject)]
+#[graphql(complex)]
```

Comment:
nit (optional): I'm not a huge fan of complex SimpleObjects, over the pattern we have elsewhere of the GraphQL type being a Scope + a wrapped native type -- mostly from a consistency perspective (spreads the implementation of fields to unfamiliar places), but this case is indeed quite borderline.

Reply:
Yep, that's right.